### PR TITLE
feat: improve handling of keyboard events on text inputs

### PR DESCRIPTION
**Pull request by kennethlynne**
This is a breaking change! 

Match the current react-native API for key press events on text inputs. See https://jsfiddle.net/kennethlynne/dL3ng4cc/ to test how different keyboard input is mapped to text

React native's API:
https://facebook.github.io/react-native/docs/textinput.html#onkeypress

Closes https://github.com/necolas/react-native-web/issues/518

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 node_modules
 dist
+dist-examples
+.idea
```
Comment (necolas) [inline .gitignore]: Please revert these changes

File: src/components/TextInput/index.js [hunk 2]
```diff
@@ -228,6 +228,7 @@ class TextInput extends Component {
       onChange: normalizeEventHandler(this._handleChange),
       onFocus: normalizeEventHandler(this._handleFocus),
       onKeyPress: normalizeEventHandler(this._handleKeyPress),
+      onKeyDown: normalizeEventHandler(this._handleKeyDown),
```
Comment (necolas) [inline src/components/TextInput/index.js]: It looks like we don't need to use `normalizeEventHandler` here or for `onKeyPress`, as we're not using the native event or passing it to any prop handlers within the component.

**Comment by kennethlynne** [comment]
It does not look like you use semver correctly, so I have no idea how to bump the package version to indicate that this is a breaking change

File: src/components/TextInput/index.js [hunk 3]
```diff
@@ -275,12 +276,42 @@ class TextInput extends Component {
...
+      // space
+      else if (e.which === 32) {
+        keyValue = ' ';
+      }
+      else {
+        // we trim to only care about ones that has a textual representation
+        if (e.shiftKey) {
+          keyValue = String.fromCharCode(e.which).trim();
+        } else {
+          keyValue = String.fromCharCode(e.which).toLowerCase().trim();
```
Comment (necolas) [inline src/components/TextInput/index.js]: I wonder if it's worth caching the results rather than recomputing each key stroke

**Comment by necolas** [comment]
> It does not look like you use semver correctly

I haven't published a stable release yet

> I have no idea how to bump the package version to indicate that this is a breaking change

That's ok as it's not your responsibility to do that in a PR

**Comment by kennethlynne** [comment]
Ok, thanks for the clarification @necolas, makes sense! :)

File: .gitignore (hunk 1 above)
Comment (kennethlynne) [inline .gitignore]: Done

File: src/components/TextInput/index.js (hunk 2 above)
Comment (kennethlynne) [inline src/components/TextInput/index.js]: Fixed

**Comment by kennethlynne** [comment]
Fixed all requested changes and successfully ran `yarn lint && yarn fmt` @necolas

**Comment by kennethlynne** [comment]
I'm working on tests now

**Comment by kennethlynne** [comment]
Tests added and all requested changes fixed  @necolas

**Comment by necolas** [comment]
Thanks!

**Comment by necolas** [comment]
I wonder if we should add support for `Shift+Enter` as that's quite commonly used in textareas on web
